forum/qa-include/pages/user-profile.php

```diff
@@ -1395,8 +1395,10 @@
         // Show other profile fields
 
         $fieldsediting = $fieldseditable && $userediting;
+        $canSeeLink = qa_is_logged_in() || $userpoints['points'] >= 500;

```

Comment:
j.w.

Reply:
Albo, skoro w obu miejscach jest taki sam kod - utworzyć metodę, w niej przypisać liczbę do zmiennej, wykonać sprawdzenie i zwrócić true/false.

Reply:
Kod jest inny, w jednym miejscu te dane juz są, w drugim pobieram punkty z bazy. Można wyciągnąć liczbę 500 jako jakąś stałą albo coś aby była wspólna i nazwana, tak chyba najprędzej

Reply:
Sorki, nie zauważyłem tego wyciągania z bazy. To może w funkcji uwzględnić wyciąganie z bazy (linia 660) na jakiś warunek?

Reply:
Pewnie można, tylko wtedy funkcja musiałaby przyjąć jako argument id usera albo liczbę punktów gdy już ją znam, co nie bardzo mi się podoba :P Albo zawsze przyjmować id usera i zawsze wysyłać zapytanie wyciągające liczbę punktów, co już wyglądałoby ładniej, ale generowało zbędne zapytanie.

Reply:
Nie bardzo to rozumiem, więc może na (pseudo) kodzie będzie łatwiej wytłumaczyć:

`function canUserSeeLink($userPoints) {
  $minPoints = 500;

  if (!$userPoints) {
    $userPoints = qa_db_select_with_pending(qa_db_user_points_selectspec($message['touserid'], true))['points'];
  }

  return $userPoints >= $minPoints;
}`

Wtedy fragment (linia 661) z pliku **format.php** przepisać na:
`
$showUrlLinks = ($options['showurllinks'] ?? false) && (qa_is_logged_in() || canUserSeeLink());
`
, a fragment (linia 1398) z pliku **user-profile.php** na:
`
$canSeeLink = qa_is_logged_in() || canUserSeeLink($userpoints['points']);
`

Jeśli nadal nie wygląda to lepiej, to się nie upieram, ale byłbym tylko za tym, żeby tę wartość liczbową punktów wrzucić gdzieś do zmiennej i z niej korzystać.

Reply:
No ok, ale wewnątrz funkcji `canUserSeeLink` używasz `$message['touserid']`, a w tej funkcji nie ma takiej zmiennej, nigdzie też jej nie przekazujesz. W związku z tym funkcja musiałaby przyjmować raz to user id, a innym razem $userPoints, co mi się trochę średnio podoba. Inny wariant to jak mówiłem, aby przyjmowała zawsze user id i zawsze pobierała punkty z bazy i na tej podstawie decydowała, ale tutaj to co ta funkcja zrobi jest tak małe, że nie wiem czy jest sens nadkładać dodatkowe zapytanie w tej części przypadków i pobierać zawsze. Samą liczbę 500 możemy wyrzucić do jakiejś stałej chociażby i użyć, z tym myślę że nie ma problemu.

Reply:
Gotowe, przeniosłem do stałej